apps/omg_watcher/test/block_getter/core_test.exs

```diff
@@ -224,168 +220,128 @@ defmodule OMG.Watcher.BlockGetter.CoreTest do
-          block_height,
-          interval,
-          synced_height,
-          maximum_number_of_pending_blocks: 5,
-          maximum_block_withholding_time_ms: 0
-        ),
-        3_000
-      )
-
-    potential_withholding = Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 0)
-    assert {:ok, state, []} = Core.handle_downloaded_block(state, potential_withholding)
 
-    potential_withholding = Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 1)
-
-    assert {{:needs_stopping, :withholding}, _, [%Event.BlockWithholding{blknum: 2000}]} =
-             Core.handle_downloaded_block(state, potential_withholding)
+    block_height
+    |> Core.init(
+      _interval = 1_000,
+      _synced_height = 1,
+      maximum_number_of_pending_blocks: 5,
+      maximum_block_withholding_time_ms: 0
+    )
+    |> Core.get_numbers_of_blocks_to_download(3_000)
+    |> assert_check([1_000, 2_000])
+    |> Core.handle_downloaded_block(Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 0))
+    |> assert_check(:ok, [])
+    |> Core.handle_downloaded_block(Core.validate_download_response({:error, :error_reason}, <<>>, 2_000, 1))
+    |> assert_check({:needs_stopping, :withholding}, [%Event.BlockWithholding{blknum: 2000}])
   end
 
   test "get_numbers_of_blocks_to_download function returns number of potential withholding block which then is canceled" do
     block_height = 0
-    interval = 1_000
-    synced_height = 1
-
-    {state, [1_000, 2_000, 3_000, 4_000]} =
-      Core.get_numbers_of_blocks_to_download(
-        Core.init(
-          block_height,
-          interval,
-          synced_height,
-          maximum_number_of_pending_blocks: 4,
-          maximum_block_withholding_time_ms: 0
-        ),
-        20_000
-      )
 
     state =
-      state
+      block_height
+      |> Core.init(
+        _interval = 1_000,
+        _synced_height = 1,
+        maximum_number_of_pending_blocks: 4,
+        maximum_block_withholding_time_ms: 0
+      )
+      |> Core.get_numbers_of_blocks_to_download(20_000)
+      |> assert_check([1_000, 2_000, 3_000, 4_000])
       |> handle_downloaded_block(%Block{number: 1_000})
       |> handle_downloaded_block(%Block{number: 2_000})
-
-    potential_withholding = Core.validate_download_response({:error, :error_reason}, <<>>, 3_000, 0)
-    assert {:ok, state, []} = Core.handle_downloaded_block(state, potential_withholding)
-
-    assert {_, [3000, 5000, 6000]} = Core.get_numbers_of_blocks_to_download(state, 20_000)
-
-    assert {:ok, state, []} = Core.handle_downloaded_block(state, {:ok, %Block{number: 3_000}})
-
-    assert {_, [5000, 6000, 7000, 8000]} = Core.get_numbers_of_blocks_to_download(state, 20_000)
+      |> Core.handle_downloaded_block(Core.validate_download_response({:error, :error_reason}, <<>>, 3_000, 0))
+      |> assert_check(:ok, [])
+
+    state
+    |> Core.get_numbers_of_blocks_to_download(20_000)
+    |> assert_check([5_000, 6_000])
```

Comment:
thanks to the huge refactor mixed with behavior changes, this is really hard to spot and makes review a nightmare. Please don't do that. Are there any more non-refactor changes to these tests? 

also this block `3_000` missing seems like exactly the bug that was fixed - a failed block download should be retried on first attempt. What is going on here?